road_to_fit/src/index.ts

```diff
@@ -1,9 +1,83 @@
 import { Hono } from 'hono'
+import {
+  createAdmin,
+  deleteAdmin,
+  getAdminById,
+  getAdmins,
+  updateAdmin,
+} from './db/queries/admin-query'
 
 const app = new Hono()
 
-app.get('/', (c) => {
-  return c.text('Hello Hono!')
+// Create a new admin
+app.post('/admin/create', async (c) => {
+  const { name, email, password, role } = await c.req.json()
+  console.log({ name, email, password, role }, 'admin data')
+
+  if (!name || !email || !password || !role) {
+    return c.json({ error: 'All fields are required' }, 400)
+  }
+
+  if (role !== 'admin' && role !== 'user') {
+    return c.json({ error: 'Invalid role' }, 401)
+  }
+
+  try {
+    await createAdmin({ name, email, password, role })
```

Comment:
why are we ignoring this promise, Can't we return the newly created id or data as a response?

Reply:
will update